PayItForward.Web/Controllers/StoriesController.cs

```diff
@@ -0,0 +1,150 @@
+﻿namespace PayItForward.Web.Controllers
+{
+    using System;
+    using System.Security.Claims;
+    using AutoMapper;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using PayItForward.Models;
+    using PayItForward.Services.Abstraction;
+    using PayItForward.Web.Models.DonationViewModels;
+    using PayItForward.Web.Models.StoryViewModels;
+
+    public class StoriesController : Controller
+    {
+        private readonly IStoriesService storiesService;
+        private readonly IDonationsService donationsService;
+        private readonly IUsersService usersService;
+        private readonly IMapper mapper;
+        private readonly IHttpContextAccessor httpaccessor;
+
+        public StoriesController(
+            IStoriesService storiesService,
+            IDonationsService donationsService,
+            IUsersService usersService,
+            IMapper mapper,
+            IHttpContextAccessor httpaccessor)
+        {
+            this.storiesService = storiesService;
+            this.donationsService = donationsService;
+            this.usersService = usersService;
+            this.mapper = mapper;
+            this.httpaccessor = httpaccessor;
+        }
+
+        [Authorize]
+        public IActionResult Details(Guid id)
+        {
+            if (id == Guid.Empty || id == null)
+            {
+                return this.RedirectToAction(actionName: "Index", controllerName: "Home");
+            }
+
+            var storyFromDb = this.storiesService.GetStoryById(id);
+            if (storyFromDb == null)
+            {
+                return this.Content("Story not found.");
+            }
+
+            var viewModel = new DetailedStoryViewModel
+            {
+                CollectedAmount = storyFromDb.CollectedAmount,
+                DateCreated = storyFromDb.CreatedOn,
+                Description = storyFromDb.Description,
+                GoalAmount = storyFromDb.GoalAmount,
+                Id = storyFromDb.Id,
+                Title = storyFromDb.Title,
+                User = storyFromDb.User
+            };
+
+            return this.View(viewModel);
+        }
+
+        public IActionResult Donate(Guid id)
```

Comment:
You can put Authorize for this action

Reply:
You can not go there if you haven't logged in before.

Reply:
Parent has authorization

Reply:
Which is the parent that has authorization?